dev-tools/ls-apis/src/system_apis.rs

```diff
@@ -259,12 +261,15 @@ impl SystemApis {
     }
 
     /// Given the client package name for an API, return the name of the server
-    /// component that provides it
-    pub fn api_producer(
-        &self,
+    /// component(s) that provide it
+    pub fn api_producers<'apis>(
+        &'apis self,
         client: &ClientPackageName,
-    ) -> Option<&ServerComponentName> {
-        self.api_producers.get(client).map(|s| &s.0)
+    ) -> impl Iterator<Item = &'apis ServerComponentName> + 'apis {
+        self.api_producers
+            .get(client)
+            .into_iter()
+            .flat_map(|producers| producers.keys())
```

Comment:
Is this the same as `.and_then(|producer_map| producer_map.keys())`?

(Not asking you to change it but for whatever reason I found using `into_iter()` on an Option + an extra flatten here more work to grok.  Want to at least check my understanding.)